### PR TITLE
Cast return value of vector.at() to void to avoid nodiscard warning in MSVC.

### LINKED ISSUE #22053: MSVC Warning C4834 in ATen
## Bug

Got this warning when used caffe2 in our application.

```
aten\core\list_inl.h(112): warning C4834: discarding return value of function with 'nodiscard' attribute
```

image

## Environment

 - PyTorch Version (e.g., 1.0): master
 - OS (e.g., Linux): Win10
 - How you installed PyTorch (`conda`, `pip`, source): source
 - Build command you used (if compiling from source): vs2017

### PR DESCRIPTION
**Pull request by xuhdev**
Fix #22053

### CODE REVIEW

File: aten/src/ATen/core/List_inl.h [hunk 1]
```diff
@@ -109,7 +109,7 @@ typename List<T>::value_type List<T>::get(size_type pos) const {
 
 template<class T>
 typename List<T>::internal_reference_type List<T>::operator[](size_type pos) const {
-  impl_->list.at(pos); // Throw the exception if it is out of range.
+  (void) impl_->list.at(pos); // Throw the exception if it is out of range.
```
Comment (xkszltl) [inline aten/src/ATen/core/List_inl.h]: should this be more "C++"? e.g. static_cast?

Comment (xuhdev) [reply aten/src/ATen/core/List_inl.h]: Sure. Updated.

**Comment by xkszltl** [comment]
LGTM

**Review by peterjc123 — APPROVED** [review]

**Pull request closed without merging**